pandas/core/indexes/datetimes.py

```diff
@@ -1127,6 +1127,21 @@ def join(self, other, how='left', level=None, return_indexers=False,
         return Index.join(this, other, how=how, level=level,
                           return_indexers=return_indexers, sort=sort)
 
+    def _tz_compare(self, other):
+        """
+        Compare time zones of DatetimeIndex.
```

Comment:
can you expound on the reason for this here (e.g. mention the directly comparing ``pytz`` of the *same* timezone is broken), but a str comparison is ok.